File: Task3C/supplier.cpp
#include "supplier.h"
#include "stock.h"
#include <sstream>

// Default constructor
Supplier::Supplier() {
    this-> name = "";
    this->ID = 0;
}

// Parameterised constructor
Supplier::Supplier(string name, int ID) {
    this->name = name;
    this->ID = ID;
}

// Getters and setters
string Supplier::getName() {
    return this->name;
}


void Supplier::setName(string name) {
    this->name = name;
}


int Supplier::getID() {
    return this->ID;
}

void Supplier::setID(int ID) {
    this->ID = ID;
}

// View supplier information
void Supplier::viewSupplierInfo() {
    cout << "Name: " << this->name << endl;
    cout << "ID: " << this->ID << endl;
}

// Save supplier information to file
void Supplier::save(ostream& out) const {
    out << name << "," << ID << "\n";
}

// Load supplier information from file
void Supplier::load(istream& in) {
    string line;
    getline(in, line);
    stringstream ss(line);

    string token;
    getline(ss, token, ',');
    name = token;

    getline(ss, token, ',');
    ID = stoi(token);
}

File: Task4A_B/task4a_b/src/main/java/com/cps2004/Supplier.java
package com.cps2004;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.ObjectNode;

public class Supplier {
    private String name;
    private int id;

    // Default constructor
    public Supplier() {
        name = "";
        id = 0;
    }

    // Parameterized constructor
    public Supplier(String name, int id) {
        this.name = name;
        this.id = id;
    }

    // Getters and setters
    public String getName() { return name; }
    public int getId() { return id; }
    public void setName(String name) { this.name = name; }
    public void setId(int id) { this.id = id; }

    // Display supplier information
    public void viewSupplierInfo() {
        System.out.println("Supplier name: " + name);
        System.out.println("Supplier ID: " + id);
    }

}
